src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.xml

```diff
@@ -20,5 +20,9 @@
             <argument type="service" id="http_client" />
         </service>
         <service id="Psr\Http\Client\ClientInterface" alias="psr18.http_client" />
+
+        <service id="traceable_http_client" class="Symfony\Bundle\FrameworkBundle\HttpClient\TraceableHttpClient" decorates="Symfony\Contracts\HttpClient\HttpClientInterface">
```

Comment:
You can have multiple clients thanks to #30419, each one should be traceable. Also these services should be registered in debug-mode only.